layers.go

```diff
@@ -137,17 +137,20 @@ func (bp *bpLayer) read() (platform.BuildpackLayerMetadata, error) {
 			return platform.BuildpackLayerMetadata{}, errors.New(msg)
 		}
 	}
-	sha, err := ioutil.ReadFile(bp.path + ".sha")
-	if err != nil {
-		if os.IsNotExist(err) {
-			return platform.BuildpackLayerMetadata{LayerMetadata: platform.LayerMetadata{SHA: ""}, LayerMetadataFile: layerMetadataFile}, nil
+	var sha string
+	if readShaFile {
+		shaBytes, err := ioutil.ReadFile(bp.path + ".sha")
+		if err != nil && !os.IsNotExist(err) {
```

Comment:
Looking at this again... if the layer.sha file does not exist we're happily returning `platform.BuildpackLayerMetadata{LayerMetadata: platform.LayerMetadata{SHA: sha}, LayerMetadataFile: layerMetadataFile}, nil` ...so why do we need to guard this logic on `if readShaFile`? I think we could just take it out.

Reply:
I agree that it should work if we delete the `if readShaFile` condition.
We can also add a comment next to the `if err != nil && !os.IsNotExist(err) {`.
But I think it's more readable the way it's now.
What do you think?

Reply:
After thinking about it again, I decided to remove the variable as you suggested and just add a clarification comment. Thanks!

Reply:
Although...maybe it's better for the future to split this function to 2 functions because most of the times when we call it, we don't even care about the sha file.

Reply:
I decided to take you advice and to remove the `readShaFile` completely from this function.
I added a comment next to the `if` condition.
Please take a look and let me know what you think.

Reply:
Great!